testkit/src/main/scala-3.0+/pureconfig/TypeStringCompat.scala

```diff
@@ -0,0 +1,17 @@
+package pureconfig
+
+import scala.reflect.ClassTag
+
+trait TypeStringCompat[T] {
+  def typeName: String
+}
+
+object TypeStringCompat {
+  // FIXME: ClassTag are weaker cases of TypeTags, which are not available in Scala 3. This is sufficient for our use
```

Comment:
How are we planning to address this in the future? Is there an alternative more strongly typed than `ClassTag` in Scala 3?

Reply:
I believe we should be able to use [`quoted.Type`](https://dotty.epfl.ch/api/scala/quoted/Type.html) for this. Unfortunately, to use `quote.Type`, we need a `QuoteContext` in scope, which involves more than a direct type class translation. Since it requires more deep changes in the places that are using it, I preferred to defer that change to when using `ClassTag`s starts becoming a problem.